spec/features/stations/index_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe 'search page nrel' do 
+  it "redirects to the /search page" do 
+    visit "/"
+
+    select "Turing"
+
+    click_button "Find Nearest Station"
+
+    expect(current_path).to eq("/search")
+  end
+
+  it "displays the nearest fuel station with its name, address, fuel type, and access times", :vcr do 
+
+    visit "/search"
```

Comment:
I believe this might have been causing the confusion. When you go directly to the search path, you bypass the form on the root page that will select the location you care about. So, you'll get to the search controller and will be passing a nil value for location down to the facade and service. That will lead the NREL call to fail and you will get the nil error for `.map`